samples/net/sockets/big_http_download/src/big_http_download.c

```diff
@@ -99,30 +109,111 @@ ssize_t sendall(int sock, const void *buf, size_t len)
 	return 0;
 }
 
-int skip_headers(int sock)
+static int parse_status(bool *redirect)
+{
+	char *ptr;
+	int code;
+
+	ptr = strstr(response, "HTTP");
+	if (ptr == NULL) {
+		return -1;
+	}
+
+	ptr = strstr(response, " ");
+	if (ptr == NULL) {
+		return -1;
+	}
+
+	ptr++;
+
+	code = atoi(ptr);
+	if (code >= 300 && code < 400) {
+		*redirect = true;
+	}
+
+	return 0;
+}
+
+static int parse_header(bool *location_found)
+{
+	char *ptr;
+
+	ptr = strstr(response, "Location");
+	if (ptr == NULL) {
+		return 0;
+	}
+
+	ptr = strstr(response, " ");
+	if (ptr == NULL) {
+		return -1;
+	}
+
+	ptr++;
+
+	strncpy(download_url, ptr, sizeof(download_url));
+
+	ptr = strstr(download_url, "\r\n");
+	if (ptr == NULL) {
+		return -1;
+	}
+
+	*ptr = '\0';
```

Comment:
Is it possible that `strncpy` will fill in `download_url` leaving no space for terminating 0, which will result in `strstr` searching past the buffer?

Reply:
Yeah, good catch, I'll fix this.

Reply:
Fixed.